app/views/contexts/summary_context.py

```diff
@@ -53,8 +74,36 @@ def summary(self, collapsible):
         }
         return context
 
-    def get_title_for_location(self, location):
-        title = self._schema.get_section(location.section_id).get("title")
+    def section_summary(self, current_location):
+        section_id = self._schema.get_section_id_for_block_id(current_location.block_id)
+        section = self._schema.get_section(section_id)
+        section_summary = self._schema.get_summary_for_section(section_id)
+        block = self._schema.get_block(current_location.block_id)
+
+        summary_context = {
+            "summary": {
+                "title": self._title_for_location(current_location),
+                "summary_type": "SectionSummary",
+                "answers_are_editable": True,
+                "collapsible": block.get("collapsible", False),
+            }
+        }
+
+        if not section_summary:
```

Comment:
`section_summary` should be named something like `custom_section_summary` as either way a section summary is being generated and returned. 

We should also consider whether some of this logic e.g. deciding if a default or custom section summary is required, is better placed in the `SectionSummary` handler.